pkg/config/nodetreemodel/config.go

```diff
@@ -225,11 +225,63 @@ func (c *ntmConfig) SetDefault(key string, value interface{}) {
 	c.setDefault(key, value)
 }
 
+func (c *ntmConfig) findPreviousSourceNode(key string, source model.Source) (Node, error) {
+	previous := source.PreviousSource()
+	for {
+		if previous == model.SourceUnknown {
+			return nil, ErrNotFound
+		}
+		tree, err := c.getTreeBySource(previous)
+		if err != nil {
+			return nil, ErrNotFound
+		}
+		node := c.leafAtPathFromNode(key, tree)
+		if _, isMissing := node.(*missingLeafImpl); !isMissing {
+			return node, nil
+		}
+		previous = previous.PreviousSource()
+	}
+}
+
 // UnsetForSource unsets a config entry for a given source
-func (c *ntmConfig) UnsetForSource(_key string, _source model.Source) {
+func (c *ntmConfig) UnsetForSource(key string, source model.Source) {
 	c.Lock()
-	c.logErrorNotImplemented("UnsetForSource")
-	c.Unlock()
+	defer c.Unlock()
+
+	// Validate that the value at the given leaf is from the source we expect
+	var node Node = c.leafAtPathFromNode(key, c.root)
+	destNode, ok := node.(LeafNode)
+	if !ok {
+		return
+	}
+	if destNode.Source() != source {
+		return
+	}
```

Comment:
We should still remove it from the dedicated tree.

Reply:
Good idea. Added this, with a bunch of tests.